jdk Files false positive/wrong line

According to javadoc `File.getParent` should be `@Nullable` and according to intellij, `Files.isDirectory(` first parameter is `@Nonnull`, javadoc there doesn't say. So it would seem to me that the wrong line is reported, or this a false positive entirely. Either `isDirectory` is going to blow, or it's going to return false, either way `findGitDir` won't even be called if `getParent()` is null.

```
/home/xeno/IdeaProjects/gradle-semver/src/main/java/com/xenoterracide/gradle/semver/SemVerPlugin.java:68: error: [NullAway] passing @Nullable parameter 'current.getParent()' where @NonNull is required
      return findGitDir(current.getParent());
                                         ^
    (see http://t.uber.com/nullaway )
```

```java
  private static boolean isGitDirHeuristic(@NonNull Path path) {
    var dotGit = path.resolve(Path.of(".git"));
    var head = dotGit.resolve(Path.of("HEAD"));
    var config = dotGit.resolve(Path.of("config"));
    var refs = dotGit.resolve(Path.of("refs"));
    var objects = dotGit.resolve(Path.of("objects"));
    return (
      Files.isDirectory(dotGit) &&
      Files.isRegularFile(head) &&
      Files.isRegularFile(config) &&
      Files.isDirectory(refs) &&
      Files.isDirectory(objects)
    );
  }

  private static Optional<Path> findGitDir(@NonNull Path current) {
    var dotGit = current.resolve(Path.of(".git"));
    if (isGitDirHeuristic(dotGit)) {
      return Optional.of(dotGit);
    } else if (Files.isDirectory(current.getParent())) {
      return findGitDir(current.getParent()); // ERROR!
    }
    return Optional.empty();
  }
```

```
com.google.errorprone:error_prone_core:2.24.1=annotationProcessor,errorprone,testAnnotationProcessor
com.uber.nullaway:nullaway:0.10.23=annotationProcessor,errorprone,testAnnotationProcessor
org.checkerframework:dataflow-nullaway:3.40.0=annotationProcessor,errorprone,testAnnotationProcessor
```

note: I tried giving parent it's own variable (`var parent = current.getParent()`), but that didn't work either.

--- a/nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java
+++ b/nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java
@@ -681,6 +681,11 @@ public class LibraryModelsHandler extends BaseNoOpHandler {
             .put(methodRef("java.util.ArrayDeque", "offer(E)"), 0)
             .put(methodRef("java.util.ArrayDeque", "push(E)"), 0)
             .put(methodRef("java.util.ArrayDeque", "<T>toArray(T[])"), 0)
+            .put(
+                methodRef(
+                    "java.nio.file.Files",
+                    "isDirectory(java.nio.file.Path,java.nio.file.LinkOption...)"),
+                0)
             .build();
 
     private static final ImmutableSetMultimap<MethodRef, Integer> NULL_IMPLIES_TRUE_PARAMETERS =

--- a/nullaway/src/test/java/com/uber/nullaway/NullAwayFrameworkTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/NullAwayFrameworkTests.java
@@ -899,4 +899,22 @@ public class NullAwayFrameworkTests extends NullAwayTestsBase {
             "}")
         .doTest();
   }
+
+  @Test
+  public void filesIsDirectory() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import java.nio.file.Files;",
+            "import java.nio.file.Path;",
+            "import org.jetbrains.annotations.Nullable;",
+            "public class Foo {",
+            "  public boolean bar(@Nullable Path p) {",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'p' where @NonNull is required",
+            "    return Files.isDirectory(p);",
+            "  }",
+            "}")
+        .doTest();
+  }
 }

